File: src/gtfs/models/StopData.py
from pydantic import BaseModel, Field


class StopData(BaseModel):
    """Internal stop representation with route references and transfers."""

    stop_id_internal: int
    stop_id_gtfs: str
    name: str
    lat: float
    lon: float
    route_ids: list[int] = Field(default_factory=list)
    transfers: list[tuple[int, int]] = Field(default_factory=list)  # (target_stop_id, walk_time)

File: src/main/kotlin/com/raptor/gtfs/models/StopData.kt
package com.raptor.gtfs.models

import kotlinx.serialization.Serializable

@Serializable
data class StopData(
    @kotlinx.serialization.SerialName("stop_id_internal")
    val stopIdInternal: Int,
    @kotlinx.serialization.SerialName("stop_id_gtfs")
    val stopIdGtfs: String,
    val name: String,
    val lat: Double,
    val lon: Double,
    val routeIds: MutableList<Int> = mutableListOf(),
    // Pair of (target_stop_id, walk_time)
    val transfers: MutableList<Pair<Int, Int>> = mutableListOf()
)
